product_barcode_generator/models/product_template.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+import uuid
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def generate_barcode(self):
+        self.ensure_one()
+        if self.to_weight:
+            seq_internal_code = self.env.ref(
+                "product_barcode_generator.seq_ean_product_internal_ref"
+            )
+            bc = ""
+            if not self.default_code:
+                rule = self.env["barcode.rule"].search(
+                    [
+                        (
+                            "name",
+                            "=",
+                            "Price Barcodes (Computed Weight) 2 Decimals",
+                        )
+                    ]
+                )[0]
+                default_code = seq_internal_code.next_by_id()
+                while self.search_count([("default_code", "=", default_code)]) > 1:
+                    default_code = seq_internal_code.next_by_id()
+                self.default_code = default_code
+            ean = "02" + self.default_code[0:5] + "000000"
+            bc = ean[0:12] + str(self.env["barcode.nomenclature"].ean_checksum(ean))
+        else:
+            rule = self.env["barcode.rule"].search(
+                [("name", "=", "product_barcode_generator")]
+            )[0]
```

Comment:
Replace this by `self.env.ref("product_barcode_generator.product_barcode_rule")`

Reply:
done